SwiftRustIntegrationTestRunner/SwiftRustIntegrationTestRunnerTests/OptionTests.swift

```diff
@@ -97,6 +97,14 @@ class OptionTests: XCTestCase {
         
         XCTAssertNil(rust_reflect_option_opaque_rust_type(nil))
     }
+
+    func testSwiftCallRustWithOptionRefOpaqueRustType() throws {
+        let val = OptTestOpaqueRustTypeRef(123)
+        let reflect = rust_reflect_option_ref_opaque_rust_type(val)
```

Comment:
I think that this might be UB at the moment, but should be easy to fix:

Let's add an extra call to `rust_reflect_option_ref_opaque_rust_type` so that we can be confident that we are returning a reference.

By calling it twice we can gain a bit more confidence that we aren't trying to free the reference.

So something along the lines of:

```swift
/// Verify that we can pass and receive an `Option<&RustType>`.
///
/// We deinitialize the first reference and create a second to confirm that
/// deinitializing the reference does not deinitialize the Rust type.
func testSwiftCallRustWithOptionRefOpaqueRustType() throws {
    let val = OptTestOpaqueRustTypeRef(123)

    let reflect = rust_reflect_option_ref_opaque_rust_type(val)
    XCTAssertEqual(reflect!.field(), 123)
    XCTAssertNil(rust_reflect_option_ref_opaque_rust_type(nil))
    let reflect = nil;

    let second_reference = rust_reflect_option_ref_opaque_rust_type(val);
    XCTAssertEqual(second_reference!.field(), 123)
}
```

---

From what I gather this test should fail. I'll comment on how to make it pass below.

Reply:
Updated so we check it twice